static/src/javascripts/projects/common/modules/onward/breaking-news.js

```diff
@@ -0,0 +1,220 @@
+    const edition = (page.edition || '').toLowerCase();
+    const section = supportedSections[page.section];
+
+    return flatten([
+        alerts
+            .filter(alert => alert.href === 'global')
+            .map(alert => alert.content),
+        alerts
+            .filter(alert => alert.href === edition)
+            .map(alert => alert.content),
+        alerts
+            .filter(alert => section && alert.href === section)
+            .map(alert => alert.content),
+    ]);
+};
+
+// keep the local alert history in sync with live alerts
+const pruneKnownAlertIDs = (alerts: Array<Alert>): Array<Alert> => {
+    // 'dismiss' this page ID, since if there's an alert for it,
+    // we don't want to show it ever
+    knownAlertIDs[page.pageId] = true;
+
+    // then remove all known alert ids that are not
+    // in the current breaking news alerts
+    knownAlertIDs = pick(knownAlertIDs, (state, id) =>
+        alerts.some(alert => alert.id === id)
+    );
+
+    storeKnownAlertIDs();
+    return alerts;
+};
+
+// don't show alerts if we've already dismissed them
+const filterAlertsByDismissed = (alerts: Array<Alert>): Array<Alert> =>
+    alerts.filter(alert => knownAlertIDs[alert.id] !== true);
+
+// don't show alerts if they're over a certain age
+const filterAlertsByAge = (alerts: Array<Alert>): Array<Alert> =>
+    alerts.filter(alert => {
+        const alertTime = alert.frontPublicationDate;
+        return (
+            alertTime &&
+            relativeDates.isWithinSeconds(new Date(alertTime), 1200)
+        ); // 20 mins
+    });
+
+// we only show one alert at a time, pick the youngest available
+const pickNewest = (alerts: Array<Alert>): Alert =>
+    alerts.sort((a, b) => b.frontPublicationDate - a.frontPublicationDate)[0];
+
+const renderAlert = (alert: Alert): bonzo => {
+    alert.marque36icon = inlineSvg('marque36icon');
+    alert.closeIcon = inlineSvg('closeCentralIcon');
+
+    const $alert = bonzo.create(template(alertHtml, alert));
+    const closeButton = $('.js-breaking-news__item__close', $alert)[0];
+
+    if (closeButton) {
+        bean.on(closeButton, 'click', () => {
+            fastdom.write(() => {
+                $('[data-breaking-article-id]').hide();
+            });
+            markAlertAsDismissed(alert.id);
+        });
+    }
+
+    return $alert;
+};
+
+const renderSpectre = ($breakingNews: bonzo): bonzo =>
+    bonzo(bonzo.create($breakingNews[0]))
+        .addClass('breaking-news--spectre')
+        .removeClass('breaking-news--fade-in breaking-news--hidden');
+
+// show an alert
+const showAlert = (alert: Alert): Alert => {
+    if (alert) {
+        const $body = bonzo(document.body);
+        const $breakingNews = bonzo(qwery('.js-breaking-news-placeholder'));
+
+        // if its the first time we've seen this alert, we wait 3 secs to show it
+        // otherwise we show it immediately
+        const alertDelay = has(knownAlertIDs, alert.id) ? 0 : DEFAULT_DELAY;
```

Comment:
Assuming we're using 'has' to check if the specified property name (alert.id) exists as a direct property of `knownAlertIds`, we could just use `hasOwnProperty` instead and drop the 'has' lodash lib? eg.

`const alertDelay = knownAlertIDs.hasOwnProperty(alert.id)  ? 0 : DEFAULT_DELAY;`

Alternatively if we're just checking to see if it has the property at all (direct or inherited) we could just check whether knownAlertIDs[alert.id] is truthy?